### ISSUE TITLE
 pidgin: Audio/Voice does not work

### ISSUE THREAD

**Issue by d4g** [NixOS/nixpkgs#32015]
## Issue description
I am unable to use the "media" options for chat, aka voice chat.

I also cannot output audio like sound effects.
If I trigger a sound effect, using the alsa output i get the following error on the debugging console:
`sound: Unable to create GStreamer audiosink.`

Adding the "Voice/Video Settings" plugin, I get the following error running the microphone test:

`(23:38:45) gtkmedia: Unable to find a suitable element for the default audio source.
(23:38:45) gtkmedia: Unable to find a suitable element for the default audio sink.
(Pidgin:13356): GStreamer-CRITICAL **: gst_bin_add_many: assertion 'GST_IS_ELEMENT (element_1)' failed
(Pidgin:13356): GStreamer-CRITICAL **: gst_element_link_many: assertion 'GST_IS_ELEMENT (element_1)' failed
(23:38:45) GLib-GObject: g_object_set: assertion 'G_IS_OBJECT (object)' failed`

Starting the 'Input and Output Settings", there is no device available for selection.

### Steps to reproduce
- Start pidgin
- Go to Tools -> Preferences -> Sound
- Try to get any sound to work
- Go to Tools -> Plugins -> Check Voice/Video Settings
- Go to Tools -> Voice/Video Settings -> Microphone Test
- The micropone does not work
![image](https://user-images.githubusercontent.com/723199/33225103-f22f6c9a-d171-11e7-845d-91673b9e4ee9.png)
- Got to Tools -> Voice/Video Settings -> Input/Output Settings
- No audio devices are selectable
![image](https://user-images.githubusercontent.com/723199/33225100-d9b034c4-d171-11e7-9925-50c0b1b3077c.png)

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.0, NixOS, 18.03pre120540.b8f7027360 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(root): `"nixos-18.03pre120540.b8f7027360"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

## Relevant Pulseaudio configuration:
`   hardware.pulseaudio = { 
    enable = true;
    package = pkgs.pulseaudioFull;
    support32Bit = true;
  };  
`

**Comment by orivej** [comment]
The `package` is not necessary, but `enable` has to be set to true.

**Comment by d4g** [comment]
Regarding puse audio: it was a copy paste error from my configuration. It is enabed. The bug of course is still there.

**Comment by d4g** [comment]
Any updates? Is this package even still maintained? Last changes are 4 years old.
@vcunat

**Comment by d4g** [comment]
The error is unchanged:
```ErrorMessage
(16:18:07) gtkmedia: Unable to find a suitable element for the default audio source.
(16:18:07) gtkmedia: Unable to find a suitable element for the default audio sink.
(16:18:07) GLib-GObject: g_object_set: assertion 'G_IS_OBJECT (object)' failed
```

**Comment by vcunat** [comment]
Well, I've put myself into `meta.maintainers` due to using pidgin a lot, but I'm afraid I'm not motivated enough to solve its audio issues.  (I'm swamped with other nix/os stuff.)  It would certainly be nice if someone else could step up and maintain it properly :-)

**Comment by vcunat** [comment]
BTW, the nixos-unstable channel has a newer pidgin version than 18.03, so that might be worth trying, but I have no idea if it might've fixed this.

**Comment by d4g** [comment]
I am running on unstable. The initial issue is 8 month old. Seems like nobody wants to step up. This makes using audio communication at all unusable, because without an audio stack pidgin can’t do voice calls. It’s not only the notification sounds that is broken, it’s everything audio related for 8 month already.

**Comment by d4g** [comment]
Yes it does. It is 8 month old, the report. The issue did not change in time and the current version has the same issue. If it is of any help, I can post the current information. But with no one being interested within 8 months, what help would it be.

**Comment by d4g** [comment]
Current version:

```
### NixOS details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.17.5, NixOS, 18.09pre145679.dae9cf6106d (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 - nix-channel: `https://nixos.org/channels/nixos-unstable`

### Device details
 - device: `20J6S0DM00 ThinkPad T470p`
 - cpu: `Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz`
 - gpu: `Intel Corporation Device 591b (rev 04)`
```

**Comment by vcunat** [comment]
Oh, I see, I was confused and thought the information was posted recently.

**Comment by manveru** [comment]
@d4g @vcunat  I think i have a fix at https://github.com/NixOS/nixpkgs/pull/59003

**Comment by vcunat** [comment]
Fixed on master and 19.03.
